crates/isograph_schema/src/create_merged_selection_set.rs

```diff
@@ -713,16 +713,22 @@ fn merge_validated_selections_into_selection_map(
                                     variant,
                                 );
                             }
-                            None => merge_non_loadable_scalar_client_field(
-                                parent_type,
-                                schema,
-                                parent_map,
-                                merge_traversal_state,
-                                newly_encountered_scalar_client_field,
-                                encountered_client_field_map,
-                                variable_context,
-                                &scalar_field_selection.arguments,
-                            ),
+                            None => match newly_encountered_scalar_client_field.variant {
+                                ClientFieldVariant::Link => {}
```

Comment:
should we merge the `link` field's selection set here? I can't actually get it to exhibit behavior in which the typename is not selected but should be, so I think that would be unnecessary. But can we rely on that behavior?

Anyway, I don't know the right answer here, happy to go with whatever you think is best

Reply:
I think we don't need to merge here, we are guaranteed to have `__typename` since cache uses two layers now

Reply:
Ok Sgtm